packages/react/src/hooks/actions/useDataStoreCreateAction.ts

```diff
@@ -42,7 +42,7 @@ export const useDataStoreCreateAction = <Model extends PersistentModel>({
         AMPLIFY_SYMBOL
       );
 
-      const item = await DataStore.save(new model(fields));
+      const item = await DataStore.save(new model(fields!));
```

Comment:
We should be able to remove the `!` operator here by updating the return type of `useTypeCastFields` to no longer include `undefined` (it never returne `undefined` afaik)

Reply:
Updated